### PR TITLE
Support ingest

### PR DESCRIPTION
**Pull request by tung-vu-td**
This PR has many changes to support sending events to Ingest endpoints, the biggest areas are:
- Use new KeenClient 4.0.0 to support sending multiple requests to Ingest endpoints.
- Support of Ingest format, such as removing Keen format special fields, as well as adding Ingest format special fields.
- Light, one time migration from old keen format to minimize data loss.
- New initialization APIs and removing dedicated `initializeApiEndpoint:` api.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,14 @@ iOS and tvOS SDK for [Treasure Data](http://www.treasuredata.com/). With this SD
 
 Also, there is an alternative SDK written in Swift [https://github.com/recruit-lifestyle/TreasureDataSDK](https://github.com/recruit-lifestyle/TreasureDataSDK). Note, however, that it does not support current GDPR functionality in the mainstream TD SDKs.
 
+## Migration to version 1
+
+Version 1 has major changes that are not backward compatible with previous versions. If you are upgrading from version 0.9.0 or earlier, your code will not run correctly without doing these following steps:
+- API endpoint has changed to Ingestion Endpoint. For now, the only available endpoint is https://us01.records.in.treasuredata.com.
```
Comment (kietdo360) [inline README.md:11]: Just a notes that we need to check with BE ingest whether this still valid , or team had provided support in all sites

Comment (tung-vu-td) [reply README.md]: Thanks for spotting this. Please check the latest commit for more general wording.

**Review by kietdo360 — APPROVED** [review]
LGTM

**Review by nmpennypacker — APPROVED** [review]
Reviewed for data exfiltrations

**Pull request merged**